### PR TITLE
Configuration for hal+json

### PR DESCRIPTION
**Pull request by baldurrensch**
## Configuration required to serialize to hal+ json

Added the ability to specify what the keys in the json serialization is for links and relations. That way, proper hal+json serialization is possible. 
## Property based relation embedding

Added a property factory that allows to make properties embedded relations. This avoids having to create services.

Please let me know if you want me to add anything. Thanks.

### CODE REVIEW

File: Factory/PropertyFactory.php [hunk 1]
```diff
@@ -0,0 +1,16 @@
...
+{
+	/**
+	 * Returns the property
+	 * @param  mixed $property
+	 * @return mixed
+	 */
+	public function retrieveProperty($property)
+	{
+		return $property;
+	}
```
Comment (stof) [inline Factory/PropertyFactory.php]: Please use spaces for the indentation, not tabs

Comment (baldurrensch) [inline Factory/PropertyFactory.php]: Argh. Of course. I'll fix that.

Comment (baldurrensch) [inline Factory/PropertyFactory.php]: DONE

**Comment by baldurrensch** [comment]
I just noticed that I forgot to update the YAML Driver for the property embedding. Will add, and update the PR.

**Comment by adrienbrault** [comment]
About the tests, can you:
- update the metadata driver tests ?
- add an environment "hal" in the functional test and verify that the output uses the configuration ?
- test the property feature in the functional tests ?

Thanks!

File: Metadata/Driver/AnnotationDriver.php [hunk 2]
```diff
@@ -41,14 +41,28 @@ public function loadMetadataForClass(\ReflectionClass $class)
...
+                            throw new \RuntimeException('The @Content provider parameters should be an array of 2 values, a service id and a method.');
+                        }
+
+                        $relationContentMetadata = new RelationContentMetadata($annotation->embed->provider[0], $annotation->embed->provider[1]);
+                        $relationMetadata->setContent($relationContentMetadata);
+                        $relationContentMetadata->setProviderArguments($annotation->embed->providerArguments ?: array());
+                    } else {
+                        $relationContentMetadata = new RelationContentMetadata("fsc_hateoas.factory.property", "retrieveProperty");
+                        $relationMetadata->setContent($relationContentMetadata);
+                        $relationContentMetadata->setProviderArguments(array('property' => $annotation->embed->property));
```
Comment (adrienbrault) [inline Metadata/Driver/AnnotationDriver.php]: The provider arguments should be a simple array and not an associative array.

**Comment by baldurrensch** [comment]
Reopened this PR as https://github.com/TheFootballSocialClub/FSCHateoasBundle/pull/10, since this PR was closed when the `dev` branch was deleted.

**Pull request closed without merging**